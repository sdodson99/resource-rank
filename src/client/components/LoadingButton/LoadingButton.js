import React from 'react';
import PropTypes from 'prop-types';
import styles from './LoadingButton.module.css';

const LoadingButton = ({ isLoading, onClick, children }) => {
  return (
    <button
      className={styles.LoadingButton}
      data-testid="LoadingButton"
      onClick={onClick}
    >
      {isLoading && <div>Loading...</div>}
      {!isLoading && children}
    </button>
  );
};

LoadingButton.propTypes = {
  isLoading: PropTypes.bool,
  onClick: PropTypes.func,
  children: PropTypes.node,
};

LoadingButton.defaultProps = {
  isLoading: false,
};

export default LoadingButton;
