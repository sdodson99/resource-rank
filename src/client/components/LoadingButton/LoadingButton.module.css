.LoadingButton {
  background-color: var(--color-secondary);

  @apply px-4 py-2 rounded transition duration-200;
}

.LoadingButton:hover {
  background-color: var(--color-secondary-dark);
}
