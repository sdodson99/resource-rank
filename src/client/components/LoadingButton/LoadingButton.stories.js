import React from 'react';
import LoadingButton from './LoadingButton';

export default {
  component: LoadingButton,
  title: 'Components/LoadingButton',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <LoadingButton {...args} />;

export const NotLoading = Template.bind({});
NotLoading.args = {
  children: 'Login',
};

export const Loading = Template.bind({});
Loading.args = {
  children: 'Login',
  isLoading: true,
};

export const Disabled = Template.bind({});
Disabled.args = {
  children: 'Login',
  disabled: true,
};
