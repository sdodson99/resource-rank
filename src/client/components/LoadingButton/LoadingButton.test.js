import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import LoadingButton from './LoadingButton';
import renderer from 'react-test-renderer';

describe('<LoadingButton />', () => {
  test('it should mount', () => {
    render(<LoadingButton />);

    const loadingButton = screen.getByTestId('LoadingButton');

    expect(loadingButton).toBeInTheDocument();
  });

  describe('with loading', () => {
    it('should render correctly', () => {
      const tree = renderer
        .create(<LoadingButton isLoading={true}>Content</LoadingButton>)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('with not loading', () => {
    it('should render correctly', () => {
      const tree = renderer
        .create(<LoadingButton isLoading={false}>Content</LoadingButton>)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
