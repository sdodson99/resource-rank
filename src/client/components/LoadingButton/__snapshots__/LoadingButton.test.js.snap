// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingButton /> with loading should render correctly 1`] = `
<button
  className="LoadingButton"
  data-testid="LoadingButton"
>
  <div>
    Loading...
  </div>
</button>
`;

exports[`<LoadingButton /> with not loading should render correctly 1`] = `
<button
  className="LoadingButton"
  data-testid="LoadingButton"
>
  Content
</button>
`;
